tools/hooks/pre-commit.js

```diff
@@ -0,0 +1,30 @@
+// Check if benchmark dependencies have explicit versions
+const semver = require("semver");
+const { EOL } = require("os");
+
+const { dependencies } = require("../../package.json");
+const { targetList } = require("../../src/cli-flags-helper");
+
+// babel -> @babel/standalone
+targetList.delete("babel");
+targetList.add("@babel/standalone");
+
+const invalid = [];
+targetList.forEach(dependency => {
+  const version = dependencies[dependency];
+  if (!semver.valid(version)) {
+    invalid.push({ dependency, version });
+  }
+});
```

Comment:
Let’s use `reduce` instead of `forEach` here.

Reply:
`targetList` is a `Set`, so `reduce` is not possible. However, we can do something like `[...targetList].reduce`.

Reply:
Right. Let’s do that — `reduce` expresses the intent more clearly IMHO.

Reply:
Done.